nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearch.java

```diff
@@ -178,8 +190,18 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                         public void process(final InputStream in) throws IOException {
                             String json = IOUtils.toString(in, charset)
                                     .replace("\r\n", " ").replace('\n', ' ').replace('\r', ' ');
-                            bulk.add(esClient.get().prepareIndex(index, docType, id)
-                                    .setSource(json.getBytes(charset)));
+
+                            if (indexOp.equalsIgnoreCase("index")) {
+                                bulk.add(esClient.get().prepareIndex(index, docType, id)
+                                        .setSource(json.getBytes(charset)));
+                            } else if (indexOp.equalsIgnoreCase("upsert")) {
+                                bulk.add(esClient.get().prepareUpdate(index, docType, id)
+                                        .setDoc(json.getBytes(charset))
+                                        .setDocAsUpsert(true));
+                            } else if (indexOp.equalsIgnoreCase("update")) {
+                                bulk.add(esClient.get().prepareUpdate(index, docType, id)
+                                        .setDoc(json.getBytes(charset)));
+                            }
```

Comment:
If the index operation is not one of these three values, it looks like the file is quietly ignored and still transferred to success. Can you add an else clause that throws an IOException, and unit test(s) to test these code paths?